crates/sail-plan/src/function/scalar/array.rs

```diff
@@ -12,13 +15,59 @@ fn array_repeat(element: expr::Expr, count: expr::Expr) -> expr::Expr {
     expr_fn::array_repeat(element, count)
 }
 
+fn array_compact(array: expr::Expr) -> expr::Expr {
+    expr_fn::array_remove_all(array, lit(ScalarValue::Null))
+}
+
+fn slice(array: expr::Expr, start: expr::Expr, length: expr::Expr) -> expr::Expr {
+    let start = expr::Expr::Cast(expr::Cast {
+        expr: Box::new(start),
+        data_type: DataType::Int64,
+    });
+    let length = expr::Expr::Cast(expr::Cast {
+        expr: Box::new(length),
+        data_type: DataType::Int64,
+    });
+    let end = expr::Expr::BinaryExpr(BinaryExpr {
+        left: Box::new(start.clone()),
+        op: Operator::Plus,
+        right: Box::new(expr::Expr::BinaryExpr(BinaryExpr {
+            left: Box::new(length),
+            op: Operator::Minus,
+            right: Box::new(lit(ScalarValue::Int64(Some(1)))),
+        })),
+    });
+    expr_fn::array_slice(array, start, end, None)
+}
+
+fn sort_array(args: Vec<expr::Expr>) -> PlanResult<expr::Expr> {
+    let (array, asc) = args.two()?;
+    let (sort, nulls) = if asc == lit(ScalarValue::Boolean(Some(true))) {
+        (
+            lit(ScalarValue::Utf8(Some("ASC".to_string()))),
+            lit(ScalarValue::Utf8(Some("NULLS FIRST".to_string()))),
+        )
+    } else if asc == lit(ScalarValue::Boolean(Some(false))) {
```

Comment:
I'd prefer `match` on `Expr::Literal(ScalarValue::Boolean(Some(true)))` etc. rather than `==`. In this way, we don't need to worry about how `Eq` or `PartialEq` behave for these types.